tests/integration/multicluster/remote/main_test.go

```diff
@@ -44,12 +45,18 @@ func TestMain(m *testing.M) {
 		RequireEnvironment(environment.Kube).
 		RequireMinClusters(2).
 		Setup(multicluster.Setup(&controlPlaneValues, &clusterLocalNS, &mcReachabilityNS)).
+		SetupOnEnv(environment.Kube, func(ctx resource.Context) error {
+			clustersByNetwork = ctx.Environment().(*kube.Environment).ClustersByNetwork()
+			return nil
+		}).
 		Setup(kube.Setup(func(s *kube.Settings) {
-			// Make all clusters use the same control plane
+			// Make all clusters in a network use the same control plane
```

Comment:
The only issue with doing this is that we're not testing cross-network discovery

Reply:
Good point. So maybe we need to expand this a bit further to have a single topology that tests everything:

- network1
  - cluster1 (control plane)
  - cluster2 (remote -> cluster1)
  - cluster3 (remote -> cluster4)
- network2
  - cluster4 (control plane)
  - cluster5 (remote -> cluster4)
  - cluster6 (remote -> cluster1)

@howardjohn do you think this would cause any issues if we used this as our standard environment for all tests?

Reply:
It may slow things down quite a bit if we leverage all 6 clusters. We'll want more parallelism in the setup scripts and the tests too, where possible.

Reply:
for everything or for multicluster tests only? I would be concerned with 6 clusters. We can scale up the pods that run the tests but I would be worried about etcd io usage even at that high number of clusters

Reply:
@howardjohn  Just trying to think about how we can test all the permutations both in single and multicluster at the same time. I think the minimum is 5 clusters:

network 1: 2 masters, 1 remote (this is all permutations for single cluster)
network 2: 1 master, 1 remote(to network1)

Maybe we just give this a try and see what the impact is on CI?

Reply:
MC tests finish in just under 29 mins and that's without https://github.com/istio/istio/pull/24523. 

If we're testing environments per-prow job maybe we don't need separate multimaster/remote suites and we can just run those tests once.

Reply:
what were they before with 3 clusters?

Reply:
Usually closer to 20. I'm hoping the PR to deploy in parallel helps but things will definitely take a long time once we start deploying in every suite.